src/engraving/libmscore/system.cpp

```diff
@@ -1808,6 +1808,53 @@ qreal System::lastNoteRestSegmentX(bool trailing)
     return margin;
 }
 
+//---------------------------------------------------------
+//   lastChordRest
+//    returns the last chordrest of a system for a particular track
+//---------------------------------------------------------
+
+ChordRest* System::lastChordRest(int track)
+{
+    for (auto measureBaseIter = measures().rbegin(); measureBaseIter != measures().rend(); measureBaseIter++) {
+        if ((*measureBaseIter)->isMeasure()) {
+            const Measure* measure = static_cast<const Measure*>(*measureBaseIter);
+            for (const Segment* seg = measure->last(); seg; seg = seg->prev()) {
+                if (seg->isChordRestType()) {
+                    ChordRest* cr = seg->cr(track);
+                    if (cr) {
+                        return cr;
+                    }
+                }
+            }
+        }
+    }
+    return nullptr;
+}
+
+//---------------------------------------------------------
+//   firstChordRest
+//    returns the last chordrest of a system for a particular track
+//---------------------------------------------------------
+
+ChordRest* System::firstChordRest(int track)
+{
+    qreal margin = score()->spatium();
+    for (const MeasureBase* mb : measures()) {
+        if (!mb->isMeasure()) {
+            continue;
+        }
+        const Measure* measure = static_cast<const Measure*>(mb);
+        for (const Segment* seg = measure->first(); seg; seg = seg->next()) {
```

Comment:
Doesn't build, at least not with MSVC: 
"first()" is not a member of 'stdvector<Ms::ChordRest *, std::allocator<Ms::ChordRest *>>'